source/elements/oneDAL/source/programming_model/computational_modes.rst

```diff
@@ -5,11 +5,18 @@ Computational modes
 -----
 Batch
 -----
-
+In the batch processing mode, the algorithm works with the entire data set to produce the final
+ result. A more complex scenario occurs when the entire data set is not available at the moment
+ or the data set does not fit into the device memory.
 ------
 Online
 ------
-
+In the online processing mode, the algorithm processes a data set in blocks streamed into device
+ memory by doing incrementally updating partial results, which are finalized upon processing of
+ the last data block.
```

Comment:
I'd suggest rewriting it into:
_In the online processing mode, the algorithm processes a data set in blocks streamed into the device's memory.
Partial results are updated incrementally and finalized when the last data block is processed._

Reply:
done